### PR TITLE
[WIP] More persistance

### PR DESCRIPTION
**Pull request by yglukhov**

### CODE REVIEW

File: beacon_chain/beacon_chain_db.nim [hunk 1]
```diff
@@ -31,23 +34,67 @@ proc hashToBlockKey(h: Eth2Digest): DbKey =
   result.data[1 .. ^1] = h.data
   result.dataEndPos = 32
 
+proc slotToBlockHashKey(s: uint64): DbKey =
+  result.data[0] = byte ord(kSlotToBlockHash)
```
Comment (mratsim) [inline beacon_chain/beacon_chain_db.nim]: `byte(kSlotToBlockHash)` directly?

File: beacon_chain/beacon_chain_db.nim [hunk 2]
```diff
@@ -31,23 +34,67 @@ proc hashToBlockKey(h: Eth2Digest): DbKey =
...
+
+proc slotToStateKey(s: uint64): DbKey =
+  result.data[0] = byte ord(kSlotToState)
+  copyMem(addr result.data[1], unsafeAddr(s), sizeof(s))
+  result.dataEndPos = uint8 sizeof(s)
+
+proc hashToValidatorRegistryChangeLogKey(deltaChainTip: Eth2Digest): DbKey =
+  result.data[0] = byte ord(kHashToValidatorRegistryChangeLog)
+  result.data[1 .. ^1] = deltaChainTip.data
+  result.dataEndPos = 32
```
Comment (mratsim) [inline beacon_chain/beacon_chain_db.nim]: It seems like `DbKey` stores both:
 - slots (uint64 and padded with zeros for [9 ..^ 33])
 - Digests (uint256)

This should be made clear in the type description. Also noting that there is realistically no chance of collision (i.e. you have more chance of your computer crashing in the next 5 seconds of reading this).

Comment (mratsim) [reply beacon_chain/beacon_chain_db.nim]: Maybe we should use a distinct type for SlotKey and DigestKey on top of DbKey?

Comment (yglukhov) [reply beacon_chain/beacon_chain_db.nim]: Note that there's no chance of collision as the first byte of the key is its type, so different mappings (slot -> block hash, slot -> hash state, hash -> block, hash -> state) will never collide, if i understood your concern. Exact same approach is taken in nimbus, and historically was inspired by py-evm which does something similar.

**Pull request merged**